### ISSUE TITLE
Patent violation

### ISSUE THREAD

**Issue by gamelaster** [lawrie/fpga_pio#9]
I just want to give heads up about the fact that even PIO is not that unique (other MCUs have similar cores as PIO) Raspberry Pi decided to really patent PIO: ☹️ 
https://patents.google.com/patent/WO2022153025A1/en

**Comment by lawrie** [comment]
I notice that one of the patent holders, Luke Wren (https://github.com/Wren6991), is watching this repository._

**Comment by ke5fx** [comment]
Sheesh, what a ridiculous patent.  Hard to see how something like Cypress's GPIF engine doesn't pre-empt at least some of the claims.

**Comment by greenscreenflicker** [comment]
Let me point out, that the parent is an application only (as seen by the A in the application). Based on my own patents, i can state, if you don't get B right away (application granted), it's a hard time to  get to B.

**Comment by gamelaster** [comment]
@greenscreenflicker which one A do you exactly mean? On the end of patent number? (I can't really find what you mean)

**Comment by greenscreenflicker** [comment]
Yes, A1 means application received by the office, B and C means granted.

**Comment by gamelaster** [comment]
https://patents.google.com/patent/EP4278267A1/en some changes happened in mean time, but even on European Patent level, it is "Pending"

**Comment by WarlockD** [comment]
The patent doesn't seem to claim to "own" a programable GPIO port but just describes its own implementation.  Its not vague enough to describe all programable PIO likes out there.  I kind of want to look up Cypress's patent on theirs to see if its similar.
